sympy/physics/continuum_mechanics/beam.py

```diff
@@ -0,0 +1,343 @@
+
+    def apply_slope_boundary_conditions(self, *s_bcs):
+        """
+        Takes only the slope boundary conditions as input.
+
+        Examples
+        ========
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import Symbol
+        >>> E = Symbol('E')
+        >>> I = Symbol('I')
+        >>> b = Beam(4, E, I)
+        >>> bcs = b.apply_boundary_conditions(moment = [(0, 4), (4, 0)], deflection = [(0, 2)], slope = [(0, 1)])
+        >>> bcs
+        {'deflection': [(0, 2)], 'moment': [(0, 4), (4, 0)], 'slope': [(0, 1)]}
+        >>> b.apply_slope_boundary_conditions((4, 3), (5, 0))
+        [(0, 1), (4, 3), (5, 0)]
+
+        """
+        for bcs in s_bcs:
+            self._boundary_conditions['slope'].append(bcs)
+        return self._boundary_conditions['slope']
+
+    def apply_deflection_boundary_conditions(self, *d_bcs):
+        """
+        Takes only the slope boundary conditions as input.
+
+        Examples
+        ========
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import Symbol
+        >>> E = Symbol('E')
+        >>> I = Symbol('I')
+        >>> b = Beam(4, E, I)
+        >>> bcs = b.apply_boundary_conditions(moment = [(0, 4), (4, 0)], deflection = [(0, 2)], slope = [(0, 1)])
+        >>> bcs
+        {'deflection': [(0, 2)], 'moment': [(0, 4), (4, 0)], 'slope': [(0, 1)]}
+        >>> b.apply_deflection_boundary_conditions((4, 3), (5, 0))
+        [(0, 2), (4, 3), (5, 0)]
+
+        """
+        for bcs in d_bcs:
+            self._boundary_conditions['deflection'].append(bcs)
+        return self._boundary_conditions['deflection']
+
+    def boundary_conditions(self):
+        """
+        Returns a dictionary of boundary conditions applied on the beam.
+        """
+        return self._boundary_conditions
+
+    def apply_loads(self, *loads):
+        """
+        Takes PointLoad and DistributedLoad as input. This method would apply
+        the loads, that are passed as arguments, to the beam object. Internally
+        this method would represent the PointLoads and DistributedLoads using
+        Singularity Functions.
+
+        Examples
+        ========
+
+        """
+
+
+class PointLoad(object):
+    """A Point Load.
+
+    A load applied to a single, specific point. It is also known as a
+    concentrated load.
+
+    Parameters
+    ==========
+    location : Point
+        The specific point of the applied load.
+    value : Sympifyable
+        A SymPy expression representing the value of the applied load.
+
+    Examples
+    ========
+    >>> from sympy.physics.continuum_mechanics.beam import PointLoad
+    >>> from sympy.physics.mechanics import Point
+    >>> p = Point('4')
+    >>> PointLoad(location = p, value = -4)
+    PointLoad(4, -4, Load)
+
+    A Moment can be defined just by passing moment=True as an argument.
+
+    >>> PointLoad(location = p, value = -4, moment=True)
```

Comment:
Do you think this API is better than:

MomentPointLoad()

and

ForcePointLoad()

Or PointForce() PointMoment()